File: LargestPalindromeProduct/LPP.java
/*
Project Euler # 4

Brute Force
*/

public class LPP {
    public static void main(String[] args) {
        int n, largest = 0;

        for ( int i=100; i<1000; i++ )
            for ( int j=100; j<1000; j++ )
                if ( is_palindrome(i*j) && i*j > largest )
                    largest = i*j;

        System.out.println(largest);
    }

    public static boolean is_palindrome(int n) {
        String a = Integer.toString(n);
        String b = new StringBuilder(a).reverse().toString();

        return a.equals(b);
    }

}

File: LargestPalindromeProduct/LPP.js
/*
Project Euler # 4

Brute Force
*/

var max = 0;

function reverse(s) {
    var array = s.split("");
    var reverseArray = array.reverse();
    return reverseArray.join("");
}

function isPalindrome(s) {
    return s === reverse(s);
}

for (var i = 100; i < 1000; i++) {
    for (var j = 100; j < 1000; j++) {
        var num = i * j;
        if (isPalindrome(num.toString()) && num > max) {
            max = num;
        }
    }
}

console.log(max);
